client/state/plugins/premium/reducer.js

```diff
@@ -0,0 +1,150 @@
+/**
+* External dependencies
+*/
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PLUGIN_SETUP_FETCH_INSTRUCTIONS,
+	PLUGIN_SETUP_RECEIVE_INSTRUCTIONS,
+	PLUGIN_SETUP_START,
+	PLUGIN_SETUP_INSTALL,
+	PLUGIN_SETUP_ACTIVATE,
+	PLUGIN_SETUP_CONFIGURE,
+	PLUGIN_SETUP_FINISH,
+	PLUGIN_SETUP_ALL_FINISH,
+	PLUGIN_SETUP_ERROR
+} from 'state/action-types';
+
+export function isRequesting( state = false, action ) {
+	switch ( action.type ) {
+		case PLUGIN_SETUP_FETCH_INSTRUCTIONS:
+			return Object.assign( {}, state, { [ action.siteId ]: true } );
+		case PLUGIN_SETUP_RECEIVE_INSTRUCTIONS:
+			return Object.assign( {}, state, { [ action.siteId ]: false } );
+		default:
+			return state;
+	}
+}
+
+function pluginStatus( state, action ) {
```

Comment:
Can we add some individual comments explaining what each one of this functions do? I'm struggling to understand how it is divided and what's each one do